clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/DeleteGoogleInternalLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,165 @@
+    if (backendService == null) {
+      GCEUtil.updateStatusAndThrowNotFoundException("Backend service $backendServiceName not found in $region for $project",
+        task, BASE_PHASE)
+    }
+
+    def healthCheckUrl = backendService.healthChecks[0]
+    def healthCheckName = GCEUtil.getLocalName(healthCheckUrl)
+    def healthCheckType = Utils.getHealthCheckType(healthCheckUrl)
+    def healthCheck = null
+    switch (healthCheckType) {
+      case "httpHealthChecks":
+        healthCheck = compute.httpHealthChecks().get(project, healthCheckName).execute()
+        break
+      case "httpsHealthChecks":
+        healthCheck = compute.httpsHealthChecks().get(project, healthCheckName).execute()
+        break
+      case "healthChecks":
+        healthCheck = compute.healthChecks().get(project, healthCheckName).execute()
+        break
+      default:
+        log.warn("Unknown health check type for health check named: ${healthCheckName}. Not queueing any batch requests.")
+        break
+    }
+    if (healthCheck == null) {
+      GCEUtil.updateStatusAndThrowNotFoundException("Health check $healthCheckName not found for $project",
+        task, BASE_PHASE)
+    }
+
+    // Now delete all the components, waiting for each delete operation to finish.
+    def timeoutSeconds = description.deleteOperationTimeoutSeconds
+
+    Operation deleteForwardingRuleOp = compute.forwardingRules().delete(project, region, forwardingRuleName).execute()
+    if (deleteForwardingRuleOp) {
+      googleOperationPoller.waitForRegionalOperation(compute, project, region, deleteForwardingRuleOp.getName(),
+        timeoutSeconds, task, "Regional forwarding rule $forwardingRuleName", BASE_PHASE)
+    }
+
+    Operation deleteBackendServiceOp = GCEUtil.deleteIfNotInUse(
+      { return compute.regionBackendServices().delete(project, region, backendServiceName).execute() },
+      "Region backend service $backendServiceName",
+      project,
+      task,
+      BASE_PHASE
+    )
+    if (deleteBackendServiceOp) {
+      googleOperationPoller.waitForRegionalOperation(compute, project, region, deleteBackendServiceOp.getName(),
+        timeoutSeconds, task, "Regional backend service $backendServiceName", BASE_PHASE)
+    }
+
+    Closure<Operation> deleteHealthCheckClosure = null
+    switch (healthCheckType) {
+      case "httpHealthChecks":
+        deleteHealthCheckClosure = { return compute.httpHealthChecks().delete(project, healthCheckName).execute() }
+        break
+      case "httpsHealthChecks":
+        deleteHealthCheckClosure = { return compute.httpsHealthChecks().delete(project, healthCheckName).execute() }
+        break
+      case "healthChecks":
+        deleteHealthCheckClosure = { return compute.healthChecks().delete(project, healthCheckName).execute() }
+        break
+      default:
+        log.warn("Unknown health check type for health check named: ${healthCheckName}. Not queueing any batch requests.")
```

Comment:
You're still calling `deleteIfNotInUse` in this case. Do you want to log this, and then let `deleteIfNotInUse` throw an error?